src/Engine/Manager/SoundManager.cpp

```diff
@@ -311,24 +316,57 @@ void SoundManager::CreateAudioEnvironment() {
                 }
             }
         }
-            
     }
 
     sAudio.FinalizeScene(NULL);
 
     // Create Environment.
     sAudio.CreateEnvironment();
+    updateLock.unlock();
 }
 
 void SoundManager::ClearKilledComponents() {
-    soundSources.ClearKilled();
-    listeners.ClearKilled();
+    const std::vector<Component::AudioMaterial*> audioMaterialVector = audioMaterials.GetAll();
```

Comment:
Couldn't this be a const reference, to avoid copying the vector?

Reply:
Yes